pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -252,15 +252,25 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, nil
 	}
 
-	// Merge in the cluster configuration, in case the administrator has updated some "downstream" fields
-	// This will also commit the change back to the apiserver.
-	if err := r.MergeClusterConfig(ctx, operConfig); err != nil {
-		log.Printf("Failed to merge the cluster configuration: %v", err)
-		r.status.SetDegraded(statusmanager.OperatorConfig, "MergeClusterConfig",
-			fmt.Sprintf("Internal error while merging cluster configuration and operator configuration: %v", err))
+	clusterConfig := &configv1.Network{}
+	if err := r.client.Default().CRClient().Get(ctx, types.NamespacedName{Name: names.CLUSTER_CONFIG}, clusterConfig); err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
 		return reconcile.Result{}, err
 	}
 
+	if !meta.IsStatusConditionPresentAndEqual(clusterConfig.Status.Conditions, names.NetworkTypeMigrationInProgress, metav1.ConditionTrue) {
```

Comment:
Please add a comment on why we don't want to merge in cluster config during live migration.
Considering that Conditions do not exist in the default API today wouldn't that `if` always result in false?

We need to add the featuregate checks for the live migration so we don't affect the default flow for now.